code/go/0chain.net/blobbercore/allocation/allocationchange.go

```diff
@@ -33,33 +34,56 @@ type AllocationChangeProcessor interface {
 }
 
 type AllocationChangeCollector struct {
-	ConnectionID      string                      `gorm:"column:connection_id;primary_key"`
-	AllocationID      string                      `gorm:"column:allocation_id"`
-	ClientID          string                      `gorm:"column:client_id"`
-	Size              int64                       `gorm:"column:size"`
-	Changes           []*AllocationChange         `gorm:"ForeignKey:connection_id;AssociationForeignKey:connection_id"`
+	ConnectionID      string                      `gorm:"column:connection_id;primaryKey"`
+	AllocationID      string                      `gorm:"column:allocation_id;size:64;not null"`
+	ClientID          string                      `gorm:"column:client_id;size:64;not null"`
+	Size              int64                       `gorm:"column:size;not null;default:0"`
+	Changes           []*AllocationChange         `gorm:"foreignKey:ConnID"`
```

Comment:
Why the foreighKey here is not `ChangeID`, isn't `ConnID` in `AllocationChange` references the AllocationChangeCollector itself?

Reply:
`ChangeID` is primary key of `allocation_changes` table. How would you use `preload` if `Changes` field has `ChangeID` to reference `allocation_changes` table? It should use field that refers to `connection_id` in `allocation_changes` table.